### PR TITLE
Refactoring router/direction strategies

### PR DESCRIPTION
**Pull request by schroedtert**
**These are still beta features and work in progress**
With this PR some of the tasks mentioned in #374, #376, #375.
First refactoring the directories for routers and direction strategies, they are now in different folders. The direction strategies are devided in walking and waiting strategies. Furthermore the separate direction strategies are now also in separate files (as from my point of view the old file got hard to read). 
Additionally FF_Trips got removed and the feature of navigating to waiting areas is now included in the normal FF router.
But I need your help if anything else important is missing. I’m open for any suggestions. Sorry for the large number of file changes, it was a use merge ... Documentation is still to come, just wanted you to have a first look beforehand.
@anna-braun it would be great if you could test it

### CODE REVIEW

**Comment by chraibi** [comment]
@schroedtert is this closed but not merged?

**Comment by schroedtert** [comment]
> @schroedtert is this closed but not merged?

Sorry, was closed by mistake

File: direction/DirectionManager.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
+//
+// Created by Tobias Schrödter on 2019-05-13.
```
Comment (zeroset) [inline direction/DirectionManager.cpp]: Please use the license header.

File: direction/DirectionManager.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+//
+// Created by Tobias Schrödter on 2019-05-13.
```
Comment (zeroset) [inline direction/DirectionManager.h]: License header please ;).

File: direction/waiting/WaitingRandom.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
+//
+// Created by Tobias Schrödter on 2019-05-14.
+//
+#pragma  once
+
+#include "WaitingStrategy.h"
+class WaitingRandom : public WaitingStrategy{
+    virtual void Init(){
+         srand(time(0));
```
Comment (zeroset) [inline direction/waiting/WaitingRandom.h]:
```suggestion
         std::srand(time(0));
```

File: direction/waiting/WaitingRandom.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
+//
+// Created by Tobias Schrödter on 2019-05-14.
+//
+#pragma  once
+
```
Comment (zeroset) [inline direction/waiting/WaitingRandom.h]:
```suggestion

#include <cstdlib>
```

File: direction/waiting/WaitingStrategy.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Point WaitingStrategy::GetTarget(Room* room, Pedestrian* ped)
+{
+     Point waitingPos = ped->GetWaitingPos();
+     Point target;
+
+     // check if waiting pos is set
+     if (waitingPos._x == std::numeric_limits<double>::max() && waitingPos._y == std::numeric_limits<double>::max()){
+          do {
+               target = GetWaitingPosition(room, ped);
+          }while(!ped->GetBuilding()->GetSubRoomByUID(ped->GetSubRoomUID())->IsInSubRoom(target));
```
Comment (zeroset) [inline direction/waiting/WaitingStrategy.cpp]: Couldn't this block the whole application? Does it also work single threaded?

File: direction/waiting/WaitingVoronoi.cpp [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+////     }//for loop
+////
+////     // for(auto x : existing_peds)
+////     //      std::cout << "-- " << x->GetPos()._x << ", " << x->GetPos()._y << "---\n";
+////     // getc(stdin);
+////
+////     //maybe not all pedestrians could find a place, requeue them in the source
+////     if(peds_without_place.size()>0)
+////          src->AddAgentsToPool(peds_without_place);
+//     return Point(0,0);
```
Comment (zeroset) [inline direction/waiting/WaitingVoronoi.cpp]: Can you remove the comment stuff?

File: direction/walking/DirectionFloorfield.cpp [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+#include "pedestrian/Pedestrian.h"
+#include "geometry/SubRoom.h"
+
+#include "router/ff_router/UnivFFviaFM.h"
+#include "router/ff_router/FloorfieldViaFM.h"
+#include "router/ff_router/ffRouter.h"
+
+#include <chrono>
+
+/// 6
```
Comment (zeroset) [inline direction/walking/DirectionFloorfield.cpp]: 6?

File: router/smoke_router/NavigationGraph.h [hunk 8]
```diff
@@ -27,8 +27,8 @@
  **/
 #pragma once
 
-#include "navigation_graph/GraphVertex.h"
-#include "navigation_graph/GraphEdge.h"
+#include "router/smoke_router/navigation_graph/GraphVertex.h"
+#include "router/smoke_router/navigation_graph/GraphEdge.h"
```
Comment (zeroset) [inline router/smoke_router/NavigationGraph.h]: Why not keep the relative path for the include?

**Comment by chraibi** [comment]
Apart from warnings, does this PR work on the demos? @schroedtert

**Comment by schroedtert** [comment]
> Apart from warnings, does this PR work on the demos? @schroedtert

No warnings anymore, demos are working except for scenario_5_ships (which also doesn’t work anymore in develop) and scenario_2_bottleneck, where seems to be some issue when using more than 1 thread. I will take a look into this.

At some point we should clean up our demos and check them more regular, if they are still working ...

**Comment by zeroset** [comment]
> At some point we should clean up our demos and check them more regular, if they are still working ...
I created an issue (#491) for this. Help is very much appreciated.

**Comment by schroedtert** [comment]
Solved the issue with scenario_2_bottleneck. If there are no objections, I will merge this PR this weekend.

**Pull request merged**